test/javascripts/shopify_app/storage_access_test.js

```diff
@@ -0,0 +1,306 @@
+suite('StorageAccessHelper', () => { 
+  let storageAccessHelper;
+  const redirectInfoStub = {
+    hasStorageAccessUrl: 'https://hasStorageAccess.com',
+    doesNotHaveStorageAccessUrl: 'https://doesNotHaveStorageAccess.com',
+    myShopifyUrl: 'https://shop1.myshopify.io',
+    home: 'https://app.io',
+  };
+
+  let contentContainer;
+  let button;
+  let redirectStub;
+
+  setup(() => {
+    window.parent.postMessage = sinon.stub();
+
+    contentContainer = document.createElement('div');
+    button = document.createElement('button');
+
+    contentContainer.setAttribute('id', 'RequestStorageAccess');
+    button.setAttribute('id', 'TriggerAllowCookiesPrompt');
+    button.setAttribute('type', 'button');
+
+    contentContainer.appendChild(button);
+    document.body.appendChild(contentContainer);
+    storageAccessHelper = new StorageAccessHelper(redirectInfoStub);
+    redirectStub = sinon.stub(ITPHelper.prototype, 'redirect');
+  });
+
+  teardown(() => {
+    document.body.removeChild(contentContainer);
+    redirectStub.restore();
+  });
+
+  suite('execute', () => {
+    test('calls setUpCookiePartitioning if ITPHelper.canPartitionCookies returns true', () => {
+      var canPartitionCookiesStub = sinon.stub(ITPHelper.prototype, 'canPartitionCookies').callsFake(() => true);
+
+      const setUpCookiePartitioningStub = sinon.stub(storageAccessHelper, 'setUpCookiePartitioning');
+
+      storageAccessHelper.execute();
+
+      sinon.assert.called(setUpCookiePartitioningStub);
+
+      canPartitionCookiesStub.restore();
+      setUpCookiePartitioningStub.restore();
```

Comment:
Look into using [sinon sandboxes](https://sinonjs.org/releases/latest/sandbox/) to eliminate the need to restore stubs.